ovs/extensions/healthcheck/volumedriver/volumedriver_health_check.py

```diff
@@ -147,79 +144,74 @@ def check_volumedrivers(logger):
+        for vp in vpools:
+            name = "ovs-healthcheck-test-{0}.raw".format(VolumedriverHealthCheck.LOCAL_ID)
+            with file_mutex('ovs-healthcheck_check-volumedrivers'):
+                if vp.guid not in VolumedriverHealthCheck.LOCAL_SR.vpools_guids:
+                    logger.skip("Skipping vPool '{0}' because it is not living here ...".format(vp.name),
+                                        'volumedriver_{0}'.format(vp.name))
+                    continue
+                try:
+                    # delete if previous vdisk with this name exists
+                    if VolumedriverHealthCheck._check_volumedriver_remove(vpool_name=vp.name, vdisk_name=name,
+                                                                          present=False):
+                        storagedriver_guid = next((storagedriver.guid for storagedriver in vp.storagedrivers
+                                                   if storagedriver.storagerouter == VolumedriverHealthCheck.LOCAL_SR))
+                        # create a new one
                         try:
-                            # delete if previous vdisk with this name exists
-                            if VolumedriverHealthCheck._check_volumedriver_remove(vpool_name=vp.name, vdisk_name=name,
-                                                                                  present=False):
-                                storagedriver_guid = next((storagedriver.guid for storagedriver in vp.storagedrivers
-                                                           if storagedriver.storagedriver_id == vp.name +
-                                                           VolumedriverHealthCheck.MACHINE_ID))
-                                # create a new one
-                                try:
-                                    volume = VolumedriverHealthCheck._check_volumedriver(name, storagedriver_guid,
-                                                                                         vp.name)
-                                except FileExistsException:
-                                    # can be ignored until fixed in framework
-                                    # https://github.com/openvstorage/framework/issues/1247
-                                    volume = True
-
-                                if volume:
-                                    # delete the recently created
-                                    time.sleep(3)
-                                    if VolumedriverHealthCheck._check_volumedriver_remove(vpool_name=vp.name,
-                                                                                          vdisk_name=name):
-                                        # working
-                                        logger.success("Volumedriver of vPool '{0}' is working fine!".format(vp.name),
+                            volume = VolumedriverHealthCheck._check_volumedriver(name, storagedriver_guid,
```

Comment:
The ```_check_volumedriver``` function should internally have this ```try - except``` statement, because everywhere you implement this function you need to mention this TODO in theory, while when having the ```try - except``` inside the function, you only need to do this once

Also, this function should be removed since it only does 1 thing actually (Once the mentioned ticket has been resolved)